### PR TITLE
[DO NOT MERGE] Jason's Dummy Example

### PR DESCRIPTION
**Pull request by jtao1**
I started making the dummy example.

### CODE REVIEW

**Comment by ahuang11** [comment]
Thanks for getting this up!

Based on the checks:
image


I think you need to add a anaconda lock file
https://anaconda-project.readthedocs.io/en/latest/user-guide/reference.html#locking-package-versions

```
Run doit validate_project_lock:olympics
.  validate_project_lock:olympics
TaskError - taskid:validate_project_lock:olympics
PythonAction Error
Traceback (most recent call last):
  File "/usr/share/miniconda3/envs/examples-gallery-manage/lib/python3.10/site-packages/doit/action.py", line 461, in execute
    returned_value = self.py_callable(*self.args, **kwargs)
  File "/home/runner/work/examples/examples/dodo.py", line 1112, in validate_project_lock
    complain(f'Missing {lock_path} file')
  File "/home/runner/work/examples/examples/dodo.py", line [17](https://github.com/holoviz-topics/examples/actions/runs/8209752833/job/22455924229?pr=368#step:7:18)7, in complain
    raise ValidationError(msg)
dodo.ValidationError: Missing olympics/anaconda-project-lock.yml file
```

**Comment by ahuang11** [comment]
I'd also add markdown cells in between each code cell to explain what you're doing in a sentence or why.

File: olympics/olympics.ipynb [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "89e3186e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "df = pd.read_csv('data/athlete_events.csv')\n",
```
Comment (ahuang11) [inline olympics/olympics.ipynb:48]: I'd suggest simply just directly reading from URL for those perusing this notebook so they don't have to set up an entire anaconda project lock to use it.
```suggestion
    "df = pd.read_csv('https://media.githubusercontent.com/media/srungta/120-years-of-olympics/master/data/athlete_events.csv')\n",
```

Comment (jtao1) [reply olympics/olympics.ipynb:48]: Do you mean I should just change the URL to the Kaggle dataset I found it from?

File: olympics/anaconda-project.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+user_fields: [examples_config]
+
+channels: [defaults]
+
+packages: &pkgs
+- notebook >=6.5.2
+- python =3.11
+- hvplot >=0.8.2
+- pandas >=1.4.2an
```
Comment (maximlt) [inline olympics/anaconda-project.yml:21]: This version `1.4.2an` looks suspicious. Why is it suffixed with `an`?

**Review by maximlt — CHANGES_REQUESTED** [review]
The CI failed with:

```
dodo.ValidationError: Env spec 'default' has changed since the lock file was last updated.
```

This means that the `anaconda-project.yml` file has been updated since the lock has last been computed. How did you re-lock the project? Did you use the new `doit lock:<project>` command? I see the lock file for linux-64 has Panel 0.14.3 which doesn't respect the project specification.

I strongly recommend running `doit validate:olympics` before committing and pushing changes to the repository, and fixing the validation errors it reports. If you don't know how, you can ask us on Discord.

File: olympics/olympics.ipynb (hunk 1 above)
Comment (maximlt) [reply olympics/olympics.ipynb:48]: > I'd suggest simply just directly reading from URL for those perusing this notebook so they don't have to set up an entire anaconda project env to use it.

Heya, this suggestion makes sense but at the same time there's some tension with how the whole system is supposed to work. Data files are meant to be declared either via the downloads section of `anaconda-project.yml` or via an intake catalog. This allows replacing them before running the tests with smaller datasets (e.g. we copy a small dataset in the local `<project>/data` folder before running the tests). Doing it this way also means the data is either downloaded upfront (download from anaconda-project.yml) or cached (intake) on AE5, every new visitor of the live notebook (and app if there's one) won't have to wait for the data to be downloaded. In this case the CSV is 42MB, not so small but also not so big.

This is a bit described in https://examples.holoviz.org/contributing.html#manage-the-data.

But yeah I also see that it doesn't make it easier for users to run the notebook by just copy/pasting snippet from it. To be fair it's a mode I haven't considered much, many projects require a non-trivial set of dependencies so users would have to go through that first barrier anyway.

So overall I'd prefer if the change was reverted and we'd stick for now to `df = pd.read_csv('data/athlete_events.csv')`.

File: olympics/anaconda-project.yml (hunk 2 above)
Comment (jtao1) [reply olympics/anaconda-project.yml:21]: Sorry that was a typo.

**Review by maximlt — COMMENTED** [review]
The example is missing a small dataset.

File: olympics/olympics.ipynb [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+    "bar_sel = bar.select(Sport=[\"Basketball\", \"Gymnastics\"]).opts(fill_color=\"darkred\")\n",
+    "(bar * bar_sel * txt.select(Sport=[\"Basketball\", \"Gymnastics\"])).opts(title=\"Average Height of each Olympic Sport, a 28 cm difference\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "3412b2d0",
+   "metadata": {},
+   "source": [
+    "We can also create a line plot between differnece the average heights of basketball and gymanstic olympians for each year. It's interesting to see that the difference has been growing wider with it peaking at 1992. "
```
Comment (ahuang11) [inline olympics/olympics.ipynb:124]:
```suggestion
    "We can also create a line plot between difference the average heights of basketball and gymanstic olympians for each year. It's interesting to see that the difference has been growing wider with it peaking at 1992. "
```

**Comment by ahuang11** [comment]
Looks mostly good to me! Maybe write a blog post off this if this doesn't get merged (or do so regardless). I'd also love to see a brief mention of using hvPlot as an exploratory tool  then mention hvPlot can be used as an explanatory tool too

Lastly, personally, I'd like to attach screenshots in PRs to help reviewers get a visual pic of the code!

**Comment by ahuang11** [comment]
For [olympics/thumbnails/olympics.png](https://github.com/holoviz-topics/examples/pull/368/files#diff-6627b5041da61380774d04a25e1fedee452df745a5b76a50526fada08df47036)

maybe use the 100m one instead? It might be more interesting (I didn't run the notebook myself yet).

File: olympics/olympics.ipynb [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a5ab6dd1",
+   "metadata": {},
+   "source": [
+    "We can also take a look at the height and weights at the same time for each medal and sport combination. We were able to overlay the plot earlier using `*` but we can also arrange the two plots together using `+`. To make them vertical we use `.cols(1)`.\n",
+    "\n",
+    "Note that the medal defaults to \"nan\" which won't display any data on the graphs, so it needs to be changed to something else. "
```
Comment (ahuang11) [inline olympics/olympics.ipynb:147]: Instead of this, you can probably get rid of NaNs ahead of time: `df.dropna(subset=["Medal"])`

**Comment by jtao1** [comment]
The blog post idea sounds great, I'll be sure to do that.

**Review by jtao1 — COMMENTED** [review]
fixed typo

File: olympics/olympics.ipynb [hunk 5]
```diff
@@ -0,0 +1,1511 @@
+{
```
Comment (ahuang11) [inline olympics/olympics.ipynb:1]: Wondering, do HoloViz examples usually clear the notebook first before pushing/merging? @maximlt

Comment (droumis) [reply olympics/olympics.ipynb:1]: Yes, I believe they should except in rare cases. It came up at the meeting today. Maxime is going to add a (pre-commit?) check to ensure the notebook outputs are cleared

**Comment by droumis** [comment]
Great work so far @jtao1 ! Can you add a bit of a summary in the PR's description and maybe even a screenshot? Maybe this will evolve into a new example later on and it will help to have an efficient and visual summary.

File: olympics/olympics.ipynb (hunk 5 above)
Comment (maximlt) [reply olympics/olympics.ipynb:1]: Unfortunately, that cannot be done in a `pre-commit` command as some notebooks on this repo must be committed with their outputs, those that can't be run on the CI (e.g. need to large datasets, compute not available, etc.). And I don't want to try to implement something that will clear these notebooks by mistake.

Instead, I'll add a new `validate` step to check that.

**Comment by maximlt** [comment]
@jtao1 can you please clear the notebook off its outputs?

**Pull request closed without merging**